drivers/mpu9150/mpu9150.c

```diff
@@ -0,0 +1,610 @@
+            i2c_release(dev->i2c_dev);
+            dev->conf.gyro_fsr = fsr;
+            break;
+        default:
+            return -2;
+    }
+
+    return 0;
+}
+
+int mpu9150_set_accel_fsr(mpu9150_t *dev, mpu9150_accel_ranges_t fsr)
+{
+    if (dev->conf.accel_fsr == fsr) {
+        return 0;
+    }
+
+    switch (fsr) {
+        case ACCEL_FSR_2G:
+        case ACCEL_FSR_4G:
+        case ACCEL_FSR_8G:
+        case ACCEL_FSR_16G:
+            if (i2c_acquire(dev->i2c_dev)) {
+                return -1;
+            }
+            i2c_write_reg(dev->i2c_dev, dev->hw_addr,
+                    MPU9150_ACCEL_CFG_REG, (char)(fsr << 3));
+            i2c_release(dev->i2c_dev);
+            dev->conf.accel_fsr = fsr;
+            break;
+        default:
+            return -2;
+    }
+
+    return 0;
+}
+
+int mpu9150_set_sample_rate(mpu9150_t *dev, uint16_t rate)
+{
+    uint8_t divider;
+
+    if ((rate < MPU9150_MIN_SAMPLE_RATE) || (rate > MPU9150_MAX_SAMPLE_RATE)) {
+        return -2;
+    }
+    else if (dev->conf.sample_rate == rate) {
+        return 0;
+    }
+
+    /* Compute divider to achieve desired sample rate and write to rate div register */
+    divider = (1000 / rate - 1);
+
+    if (i2c_acquire(dev->i2c_dev)) {
+        return -1;
+    }
+    i2c_write_reg(dev->i2c_dev, dev->hw_addr, MPU9150_RATE_DIV_REG, (char) divider);
+
+    /* Store configured sample rate */
+    dev->conf.sample_rate = 1000 / (((uint16_t) divider) + 1);
+
+    /* Always set LPF to a maximum of half the configured sampling rate */
+    conf_lpf(dev, (dev->conf.sample_rate >> 1));
+    i2c_release(dev->i2c_dev);
+
+    return 0;
+}
+
+int mpu9150_set_compass_sample_rate(mpu9150_t *dev, uint8_t rate)
+{
+    uint8_t divider;
+
+    if ((rate < MPU9150_MIN_COMP_SMPL_RATE) || (rate > MPU9150_MAX_COMP_SMPL_RATE)
+            || (rate > dev->conf.sample_rate)) {
+        return -2;
+    }
+    else if (dev->conf.compass_sample_rate == rate) {
+        return 0;
+    }
+
+    /* Compute divider to achieve desired sample rate and write to slave ctrl register */
+    divider = (dev->conf.sample_rate / rate - 1);
+
+    if (i2c_acquire(dev->i2c_dev)) {
+        return -1;
+    }
+    i2c_write_reg(dev->i2c_dev, dev->hw_addr, MPU9150_SLAVE4_CTRL_REG, (char) divider);
+    i2c_release(dev->i2c_dev);
+
+    /* Store configured sample rate */
+    dev->conf.compass_sample_rate = dev->conf.sample_rate / (((uint16_t) divider) + 1);
+
+    return 0;
+}
+
+/*------------------------------------------------------------------------------------*/
+/*                                Internal functions                                  */
+/*------------------------------------------------------------------------------------*/
+
+/**
+ * Initialize compass
+ */
+static int compass_init(mpu9150_t *dev)
+{
+    char data[3];
+
+    /* Enable Bypass Mode to speak to compass directly */
+    conf_bypass(dev, 1);
+
+    /* Check whether compass answers correctly */
```

Comment:
I'd argue exactly the other way around. If somebody else maintains this driver who doesn't have this in the back of his head this could easily lead to not exclusive bus accesses. IMHO. If you insist on this, please add a comment stating these functions are not handling mutual exclusion.